### PR TITLE
fix: broken search

### LINKED ISSUE #2800: Search bar is broken (again?)
#### Step to reproduce
- go to `/rmrk/faq/` or any page that doesn't show nfts
- type in searchbar
- press enter or click data
- `?search=` is append on url

#### Expected Behavior 
- redirect user to search result

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2800 
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [x] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [x] I've tested PR on mobile and everything seems works
- [x] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: components/rmrk/Gallery/Search/SearchBar.vue [hunk 1]
```diff
@@ -399,7 +401,7 @@ export default class SearchBar extends mixins(
   }
 
   redirectToGalleryPageIfNeed() {
-    if (this.$route.name === 'index') {
```
Comment (Jarsen136) [inline components/rmrk/Gallery/Search/SearchBar.vue:402]: we need to support 'rmrk' 'westend', so checking `$route.name` is not a property way and I use `$route.path` here.

**Comment by Jarsen136** [comment]
> It fix #2800 but I found another thing
> 
> 
> 
> #### Step to reproduce
> 
> - Go to `/rmrk/faq`
> 
> - Search `driver` (or anything)
> 
> - Press enter
> 
> - missing the query in url (something like ?search=driver&page=1)

I will check it

**Review by roiLeo — APPROVED** [review]
lgtm

**Comment by roiLeo** [comment]
pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 151.6 USD/KSM ~ 0.33 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x07ef8d886aa719716d8f34b714dc8eacd364eb5f9c0f7d0679f2641cda9ea2d6](https://kusama.subscan.io/extrinsic/0x07ef8d886aa719716d8f34b714dc8eacd364eb5f9c0f7d0679f2641cda9ea2d6)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**